### DISCUSSION TITLE
There is no way to unload dynamic types [12606]

### DISCUSSION THREAD

**Discussion by ghost** [eProsima/Fast-DDS#3738] [Ideas]
The function `XMLProfileManager::DeleteInstance()` does not clear `dynamic_types_` member.

## Expected Behavior
It should be possible to unload dynamic types using `XMLProfileManager::DeleteInstance()`.

## Current Behavior
There is no way to unload dynamic types. The function `XMLProfileManager::DeleteInstance()` does not delete the information.

## Steps to Reproduce
1. Load a dynamic type using `XMLProfileManager::loadXMLFile()`.
2. Try to unload the dynamic type using `XMLProfileManager::DeleteInstance()`.

## System information
- Fast-RTPS version: 2.4.0
- OS: Ubuntu 20.04

**Comment by MiguelCompany** [comment]
The function `XMLProfileManager::loadXMLDynamicTypes()` is a helper that loads the type defined on the XML into `DynamicTypeBuilderFactory`. You could use `DynamicTypeBuilderFactory::delete_instance()` to achieve what you expect.

Note that `XMLProfileManager::DeleteInstance()` cannot call `DynamicTypeBuilderFactory::delete_instance()`, as the user could have used `DynamicTypeBuilderFactory` directly (without using `XMLProfileManager`)

**Comment by ghost** [comment]
My bad. I meant function `XMLProfileManager::loadXMLFile()`. I would like to use XML profiles for dynamic types, but it seems not possible to unload such profile.

**Comment by MiguelCompany** [comment]
Oh! I see. Seems like adding `dynamic_types_.clear()` to `XMLProfileManager::DeleteInstance()` would do the trick.

Now that I look at it, I think that method is missing other things, like restoring other XML loaded information to the default values. We'll try to allocate some time to fix everything on a single PR.

**Comment by chunyisong** [comment]
@MiguelCompany I wonder if there is a way by DomainParticipantFactory to reload any (not just dynamic types) profiles from  a different  xml and delete loaded profiles avoiding Conflict.

**Comment by JLBuenoLopez** [comment]
Hi @chunyisong

It is not possible to unload profiles, though you can load different XML profiles files using [`DomainParticipant::load_XML_profiles_file`](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/dds_pim/domain/domainparticipantfactory.html#_CPPv4N8eprosima7fastdds3dds24DomainParticipantFactory22load_XML_profiles_fileERKNSt6stringE)

**Comment by JLBuenoLopez** [comment]
I am labeling this ticket as enhancement and moving it to the proper discussion forum according to [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).
